### DISCUSSION TITLE
If anyone is trying to use Azure via Github Models

### DISCUSSION THREAD

**Discussion by curiosityz** [microsoft/autogen#4297] [General]
This is what worked for me:

async def main() -> None:
    # Define an agent
    weather_agent = AssistantAgent(
        name="weather_agent",
        model_client=AzureOpenAIChatCompletionClient(
    model="gpt-4o",
    api_version="2024-02-01",
    azure_endpoint="https://models.inference.ai.azure.com",
    model_capabilities={
        "vision":True,
        "function_calling":True,
        "json_output":True,
    }
),
        tools=[get_weather],
    )![Image](https://github.com/user-attachments/assets/5331d3fb-a04b-4baa-b8e3-d0dd73a70317)

**Comment by jackgerrits** [comment]
Good callout! OpenAI models on the models marketplace on GitHub support the OpenAI api, so it seems like the azure openai client also works.
